pastes/views.py

```diff
@@ -1,3 +1,45 @@
+import json
 from django.shortcuts import render
+from django.http import HttpResponse
+from django.shortcuts import redirect
 
-# Create your views here.
+from .models import Pastes
+from . import tokens
+
+def list(request):
+    p = Pastes.objects.filter(user=request.user.id)
+    return render(request, 'pastes.html', {'pastes': p})
+
+def delete(request, id):
+    p = Pastes.objects.filter(id=id).delete()
+    return redirect('/')
+    
+
+def show(request, token):
+    if token.isdigit():
+        p = Pastes.objects.filter(id=token).values('paste')
+        if p is not None:
```

Comment:
Are you sure this works? `p` at this point is still a `QuerySet`.

Did you forget to call `.first()` ?

Either way, there's two clear paths through this view where it doesn't return anything.
Also, both the if and else end the same - a chance to refactor our shared code.

Reply:
It looks like he calls `p[0]` right below this, but probably better to call `first` IMO.